src/os.rs

```diff
@@ -289,6 +292,51 @@ mod imp {
     }
 }
 
+#[cfg(target_os = "openbsd")]
+mod imp {
+    extern crate libc;
+
+    extern "C" {
+        pub fn getentropy(buf: *mut libc::c_void, buflen: libc::size_t) -> libc::c_int;
+    }
+
+    use std::io;
+    use Rng;
+
+    use super::{next_u32, next_u64};
+
+    pub struct OsRng;
+
+    impl OsRng {
+        pub fn new() -> io::Result<OsRng> {
+            Ok(OsRng)
+        }
+    }
+
+    impl Rng for OsRng {
+        fn next_u32(&mut self) -> u32 {
+            next_u32(&mut |v| self.fill_bytes(v))
+        }
+        fn next_u64(&mut self) -> u64 {
+            next_u64(&mut |v| self.fill_bytes(v))
+        }
+        fn fill_bytes(&mut self, v: &mut [u8]) {
+            // getentropy allows 256 byte chunks
```

Comment:
the comment is a bit misleading: the 256 bytes chunks is the **maximum** buffer size permitted by the syscall.